app/models/wise_transfer.rb

```diff
@@ -0,0 +1,208 @@
+      date: created_at
+    )
+  end
+
+  validates_presence_of :payment_for, :recipient_name, :recipient_email
+  validates :recipient_email, format: { with: URI::MailTo::EMAIL_REGEXP, message: "must be a valid email address" }
+  normalizes :recipient_email, with: ->(recipient_email) { recipient_email.strip.downcase }
+
+  aasm timestamps: true, whiny_persistence: true do
+    state :pending, initial: true
+    state :approved
+    state :rejected
+    state :sent
+    state :deposited
+    state :failed
+
+    event :mark_approved do
+      transitions from: :pending, to: :approved
+    end
+
+    event :mark_rejected do
+      after do
+        canonical_pending_transaction.decline!
+      end
+      transitions from: [:pending, :approved], to: :rejected
+    end
+
+    event :mark_sent do
+      after do
+        canonical_pending_transaction.update(amount_cents: -usd_amount_cents)
+      end
+      transitions from: [:approved], to: :sent
+    end
+
+    event :mark_deposited do
+      transitions from: :sent, to: :deposited
+    end
+  end
+
+  validates :amount_cents, numericality: { greater_than: 0, message: "must be positive!" }
+
+  alias_attribute :name, :recipient_name
+
+  def hcb_code
+    "HCB-#{TransactionGroupingEngine::Calculate::HcbCode::WISE_TRANSFER_CODE}-#{id}"
+  end
+
+  def admin_dropdown_description
+    "#{usd_amount.format} (#{Money.from_cents(amount_cents, currency).format} #{currency}) to #{recipient_name} (#{recipient_email}) from #{event.name}"
+  end
+
+  def local_hcb_code
+    return nil unless persisted?
+
+    @local_hcb_code ||= HcbCode.find_or_create_by(hcb_code:)
+  end
+
+  def status_color
+    if pending?
+      :warning
+    elsif approved?
+      :blue
+    elsif sent?
+      :purple
+    elsif rejected? || failed?
+      :error
+    elsif deposited?
+      :success
+    else
+      :muted
+    end
+  end
+  alias_method :state_color, :status_color
+
+  def state_text
+    aasm_state.humanize
+  end
+
+  def last_user_change_to(...)
+    user_id = versions.where_object_changes_to(...).last&.whodunnit
+
+    user_id && User.find(user_id)
+  end
+
+  def self.generate_quote(money)
+    conn = Faraday.new url: "https://api.wise.com" do |f|
+      f.request :json
+      f.response :raise_error
+      f.response :json
+    end
+
+    response = conn.post("/v3/quotes", {
+                           sourceCurrency: "USD",
+                           targetCurrency: money.currency_as_string,
+                           targetAmount: money.dollars
+                         })
+
+    payment_option = response.body["paymentOptions"].first
+    price_after_fees = Money.from_dollars(payment_option["sourceAmount"], "USD")
+    fees = payment_option["price"]["items"]
+    payin_fee = Money.from_dollars(fees.find { |fee| fee["type"] == "PAYIN" }["value"]["amount"], "USD")
+
+    price_before_payin_fee = price_after_fees - payin_fee
```

Comment:
```suggestion
    pay_in_fee = Money.from_dollars(fees.find { |fee| fee["type"] == "PAYIN" }["value"]["amount"], "USD")

    price_before_pay_in_fee = price_after_fees - pay_in_fee
```
nit because I read this as "paying" fee at first and Wise seems to treat it as two words: `option.payIn`

Reply:
ahhh camel case in a Ruby app is so annoying. I think I skipped the underscore because they refer to it as `PAYIN` in all caps without an underscore, but I agree that it's much cleaner with one.